<a href="https://colab.research.google.com/github/kevincruzortiz123-lgtm/programacion-2/blob/main/quiz26marzo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [134]:
%%writefile pedirnotas.py
def pedir_nombre():
    return input("ingrese el nombre del estudiante: ")
def pedir_notas(n1,n2,n3,n4):
    n1=float(input("nota 1: "))
    n2=float(input("nota 2: "))
    n3=float(input("nota 3: "))
    n4=float(input("nota 4: "))
    return [n1,n2,n3,n4]

Overwriting pedirnotas.py


In [135]:
%%writefile promedio.py
import statistics
def calcular_promedio(n1,n2,n3,n4):
  return (n1 + n2 + n3 + n4) /4

def mediana(n1,n2,n3,n4):
  return statistics.median([n1,n2,n3,n4])

def moda(n1,n2,n3,n4):
  return statistics.mode([n1,n2,n3,n4])

Overwriting promedio.py


In [136]:
import pedirnotas
import promedio
importlib.reload(pedirnotas)
importlib.reload(promedio)


<module 'promedio' from '/content/promedio.py'>

In [138]:
nombre = pedirnotas.pedir_nombre()
notas= pedirnotas.pedir_notas("n1","n2","n3","n4")


ingrese el nombre del estudiante: kevin
nota 1: 3.9
nota 2: 3.9
nota 3: 4.0
nota 4: 4.2


In [139]:
n1, n2, n3, n4 = notas
prom=promedio.calcular_promedio(n1,n2,n3,n4)
median=promedio.mediana(n1,n2,n3,n4)
mod=promedio.moda(n1,n2,n3,n4)

In [140]:
print("estudiente:", nombre)
print("promedio:", prom)
print("mediana",median)
print("moda:", mod)
if prom >= 3:
  print("aprobo")
else:
  print("reprobo")

estudiente: kevin
promedio: 4.0
mediana 3.95
moda: 3.9
aprobo


nota 1: 3.9
nota 2: 3.9
nota 3: 4.0
nota 4: 4.2

En promedio el estudiente obtuvo 4.05
El valor que divide el 50% de las calificaciones es  3.95
La calificacion mas repetida es 3.9
el desempeño es bueno ya que llego a sacar mas nota que la minima, esto hace que su nota sea buena y si hay mas esfuerzo por parte del estudiante pueda llegar a ser excelente


